src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -30,39 +36,20 @@ class TaskReplaceActor(
     val eventBus: EventStream,
     val readinessCheckExecutor: ReadinessCheckExecutor,
     val runSpec: RunSpec,
-    promise: Promise[Unit]) extends Actor with Stash with ReadinessBehavior with StrictLogging {
+    promise: Promise[Unit]) extends Actor with Stash with StrictLogging {
   import TaskReplaceActor._
 
-  // compute all values ====================================================================================
+  val pathId: PathId = runSpec.id
 
   // All running instances of this app
-  //
-  // Killed resident tasks are not expunged from the instances list. Ignore
-  // them. LaunchQueue takes care of launching instances against reservations
-  // first
-  val currentInstances = instanceTracker.specInstancesSync(runSpec.id).filter(_.state.goal == Goal.Running)
-
-  // In case previous master was abdicated while the deployment was still running we might have
-  // already started some new tasks.
-  // All already started and active tasks are filtered while the rest is considered
-  private[this] val (instancesAlreadyStarted, instancesToRemove) = {
-    currentInstances.partition(_.runSpecVersion == runSpec.version)
+  val instances: mutable.Map[Instance.Id, Instance] = instanceTracker.specInstancesSync(runSpec.id).map { i => i.instanceId -> i }(collection.breakOut)
+  val instancesHealth: mutable.Map[Instance.Id, Boolean] = instances.collect {
+    case (id, instance) => id -> instance.state.healthy.getOrElse(false)
   }
+  val instancesReady: mutable.Map[Instance.Id, Boolean] = mutable.Map.empty
```

Comment:
Ideally we would keep everything in the instances map.

Reply:
do you think it makes sense to try to move readiness checks out of the orchestration layer? I actually think it will help to keep the orchestrators much more tidy...